packages/permission-controller/src/Caveat.ts

```diff
@@ -82,15 +82,12 @@ export type CaveatDecorator<ParentCaveat extends CaveatConstraint> = (
  * @template Decorator - The {@link CaveatDecorator} to extract a caveat value
  * type from.
  */
-// TODO: Replace `any` with type
```

Comment:
Same, why can't this be typed?

Reply:
This usage of `any`, in a constraint, is essentially saying "the generic parameter `Decorator` needs to extend the `CaveatDecorator` type, irrespective of its generic parameter". We can't use `CaveatDecorator<unknown>`, because unknown violates the constraints on its generic parameter. The `Decorator` type still needs to be a valid `CaveatDecorator`, so it is that which should use something more specific than `any`, which is what we do.